dask_expr/_expr.py

```diff
@@ -1275,6 +1276,196 @@ def _task(self, index: int):
             )
 
 
+class MapOverlap(MapPartitions):
```

Comment:
Should we keep this in a new `dask_expr/_rolling.py`?

Reply:
Hm I'd rather not, this is not rolling specific (fillna uses this as well for example)

Reply:
Okay cool. I noticed that the `dask.dataframe` version was in a `rolling` module, so I just wanted to check.

Reply:
Yeah that part confused me :)